grpc/src/main/proto/tree_service.proto

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+syntax = "proto3";
+
+package org.projectnessie.proto;
+
+import "google/protobuf/timestamp.proto";
+import "config_service.proto";
+import "contents_service.proto";
+
+option java_multiple_files = true;
+option java_package = "org.projectnessie.api.grpc";
+option java_outer_classname = "TreeServiceProto";
+
+service TreeService {
+  rpc getAllReferences (Empty) returns (GetAllReferencesResponse) {}
+  rpc getReferenceByName (GetReferenceByNameRequest) returns (Reference) {}
+  rpc createReference (CreateReferenceRequest) returns (Reference) {}
+  rpc getDefaultBranch (Empty) returns (Reference) {}
+  rpc assignTag (AssignReferenceRequest) returns (Empty) {}
+  rpc deleteTag (DeleteReferenceRequest) returns (Empty) {}
+  rpc assignBranch (AssignReferenceRequest) returns (Empty) {}
+  rpc deleteBranch (DeleteReferenceRequest) returns (Empty) {}
+  rpc getCommitLog (CommitLogRequest) returns (CommitLogResponse) {}
+  rpc getEntries (EntriesRequest) returns (EntriesResponse) {}
+  rpc transplantCommitsIntoBranch (TransplantRequest) returns (Empty) {}
+  rpc mergeRefIntoBranch (MergeRequest) returns (Empty) {}
+  rpc commitMultipleOperations (CommitRequest) returns (Branch) {}
+}
+
+message Reference {
+  oneof type {
+    Branch branch = 1;
+    Tag tag = 2;
+    Hash hash = 3;
+  }
+}
+
+message Branch {
+  string name = 1;
+  string hash = 2;
+}
+
+message Tag {
+  string name = 1;
+  string hash = 2;
+}
+
+message Hash {
+  string hash = 1;
+}
+
+message CreateReferenceRequest {
+  string sourceRefName = 1;
+  Reference reference = 2;
+}
+
+message GetAllReferencesResponse {
+  repeated Reference reference = 1;
+}
+
+message GetReferenceByNameRequest {
+  string namedRef = 1;
+}
+
+message AssignReferenceRequest {
+  string namedRef = 1;
+  string oldHash = 2;
+  oneof assignTo {
+    Branch branch = 3;
+    Tag tag = 4;
+  }
+}
+
+message DeleteReferenceRequest {
+  string namedRef = 1;
+  string hash = 2;
```

Comment:
Could this be a simple `Reference` objects (assuming we deny deleting plain `Hash` references)?